pkg/k8s/client.go

```diff
@@ -0,0 +1,76 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+type Client struct {
+	Clientset kubernetes.Interface
+}
+
+func (c Client) ApplySecret(ctx context.Context, secret *apiv1.Secret) error {
+	_, err := c.createSecret(ctx, secret)
+	if err != nil {
+		log.Print(err)
```

Comment:
Do we want to have these log messages left in (here and throughout)? If we do, what are your thoughts on using a leveled logger and having debug logs that can be enabled or disabled?

Reply:
oof I think I accidentally left this here for debugging purposes.  I don't think we want to log errors at the package level (unless the package accepts a structured logger), my preference would be to return useful errors to the caller and let the caller decide what to do with them.